src/page-multi-facility/FacilityRow.tsx

```diff
@@ -15,25 +18,34 @@ const groupStatus = {
   infectious: true,
 };
 
-const FacilityRow: React.FC = () => {
-  const facility = useContext(FacilityContext);
+interface Props {
+  facility: Facility;
+}
+
+const FacilityRow: React.FC<Props> = ({ facility }) => {
   const confirmedCases = totalConfirmedCases(useEpidemicModelState());
-  if (!facility) {
-    throw new Error("Facility must be provided to the FacilityContext");
-  }
+  const history = useHistory();
+  const { setFacility } = useContext(FacilityContext);
+
+  const { id, name, updatedAt } = facility;
 
-  const { name } = facility;
+  const openFacilityPage = (event: React.MouseEvent<HTMLElement>) => {
+    setFacility(facility);
+    history.push("/multi-facility/facility");
```

Comment:
This is the key code to getting this all to work.  When a facility row is clicked, we set that information in a globally accessible context.  Then when we load the multi-facility page that facility is available to be used in the form in an update fashion.  If a facility is not present on the facility page, then we can create a new facility in the empty form.